### PR TITLE
Update state.md to new react website

### PR DESCRIPTION
**Pull request by demurthelemur**
Changed old react.org website link to new react.dev website

### CODE REVIEW

File: docs/state.md [hunk 1]
```diff
@@ -100,4 +100,4 @@ In a real application, you probably won't be setting state with a timer. You mig
 
 When setState is called, BlinkApp will re-render its Component. By calling setState within the Timer, the component will re-render every time the Timer ticks.
 
-State works the same way as it does in React, so for more details on handling state, you can look at the [React.Component API](https://reactjs.org/docs/react-component.html#setstate). At this point, you may have noticed that most of our examples use the default text color. To customize the text color, you will have to [learn about Style](style.md).
+State works the same way as it does in React, so for more details on handling state, you can look at the [React.Component API](https://react.dev/reference/react/Component). At this point, you may have noticed that most of our examples use the default text color. To customize the text color, you will have to [learn about Style](style.md).
```
Comment (Simek) [inline docs/state.md:103]: Hello @demurthelemur, thanks for the correction! 👍 

Let's use the hash link to lead user to the `setState` directly:

```suggestion
State works the same way as it does in React, so for more details on handling state, you can look at the [React.Component API](https://react.dev/reference/react/Component#setstate). At this point, you may have noticed that most of our examples use the default text color. To customize the text color, you will have to [learn about Style](style.md).
```

And one more question, can you backport this correction to the few latest's, versioned docs?
* https://github.com/facebook/react-native-website/tree/main/website/versioned_docs

Comment (demurthelemur) [reply docs/state.md]: I added your first suggestion but I have no clue what " backport this correction to the few latest's, versioned docs?" means :D Sorry new to this whole thing

Comment (Simek) [reply docs/state.md]: @demurthelemur With each new stable version released we copy over the `docs` directory content to `versioned_docs` to have the stable and correct documentation for a given version.

The question is - could you apply the same change to the `state.md` files located in `versioned_docs` directory? Would be nice to correct this also for 0.73 and 0.72 releases:
* https://github.com/facebook/react-native-website/blob/main/website/versioned_docs/version-0.73/state.md
* https://github.com/facebook/react-native-website/blob/main/website/versioned_docs/version-0.72/state.md

Also, please remember to accept the CLA, it's required to merge your changes in.

Comment (demurthelemur) [reply docs/state.md]: Accepted the CLA waiting for it to go through. Did the changes you asked. Thanks for the help

Comment (Simek) [reply docs/state.md]: Thanks for the corrections!

**Review by Simek — APPROVED** [review]
Thank you for the correction and backports, LGTM!

**Pull request merged**